src/Service.Tests/Unittests/MultiSourceQueryExecutionUnitTests.cs

```diff
@@ -146,15 +121,108 @@ public async Task TestMultiSourceQuery()
             QueryResult queryResult = (QueryResult)result;
             Assert.IsNotNull(queryResult.Data, "Data should be returned for multisource query.");
             IReadOnlyDictionary<string, object> data = queryResult.Data;
-            Assert.IsTrue(data.TryGetValue(queryName1, out object queryNode1), $"Query node for {queryName1} should have data populated.");
-            Assert.IsTrue(data.TryGetValue(queryName2, out object queryNode2), $"Query node for {queryName2} should have data populated.");
+            Assert.IsTrue(data.TryGetValue(QUERYNAME1, out object queryNode1), $"Query node for {QUERYNAME1} should have data populated.");
+            Assert.IsTrue(data.TryGetValue(QUERYNAME2, out object queryNode2), $"Query node for {QUERYNAME2} should have data populated.");
 
             ResultMap queryMap1 = (ResultMap)queryNode1;
             ResultMap queryMap2 = (ResultMap)queryNode2;
 
             // validate that the data returend for the queries we did matches the moq data we set up for the respective query engines.
-            Assert.AreEqual("db1", queryMap1[0].Value, $"Data returned for {queryName1} is incorrect for multi-source query");
-            Assert.AreEqual("db2", queryMap2[0].Value, $"Data returned for {queryName2} is incorrect for multi-source query");
+            Assert.AreEqual("db1", queryMap1[0].Value, $"Data returned for {QUERYNAME1} is incorrect for multi-source query");
+            Assert.AreEqual("db2", queryMap2[0].Value, $"Data returned for {QUERYNAME2} is incorrect for multi-source query");
+        }
+
+        [TestMethod]
+        public async Task TestMultiSourceQueryRest()
```

Comment:
nit: clarify in a summary what you're validating. e.g.
```csharp
/// Validates that the REST endpoint returns data from the proper datasource configured for an entity:
/// Request: /api/entity1 | Entity: Entity1 | DataSource: CosmosDB
/// Request: /api/entity2 | Entity: Entity2 | DataSource: MsSql
```

Reply:
Added.

Reply:
nit: below describes test in more explicit terms.
```csharp
/// Validates that the REST endpoint returns data from the proper datasource configured for an entity:
/// Request: /api/entity1 -> Resolves data from: Entity: Entity1 | DataSource: CosmosDB | QueryEngine: CosmosQueryEngine
/// Request: /api/entity2 -> Resolves data from: Entity: Entity2 | DataSource: MsSql    | QueryEngine: MsSqlQueryEngine
```